### DISCUSSION TITLE
Forming OCHLV data from bid/ask streams

### DISCUSSION THREAD

**Discussion by jjphung** [hootnot/oanda-api-v20#180] [General]
First off, this is such a useful library and I appreciate all the work @hootnot put into this!

Moving onto the discussion/question, does anyone have ideas on how to calculate Open, High, and Low from the streaming tick bid/ask? I noticed that in my research, people take the average of the bid and ask, like in the sibling repo `oandapyV20-examples` in order to calculate Close. However in the incoming Oanda streams, I don't think there are bids and ask prices for Open, High, and Low [[1](https://www.reddit.com/r/algotrading/comments/aawvoq/how_to_use_the_bidask_ohlc/)]. Thank you!

1. Reddit Link explaining averaging to calculate OCHLV data

**Comment by hootnot** [comment]
Hi @jjphung ,

This question was asked recently also: https://github.com/hootnot/oanda-api-v20/issues/178

Bid/Ask price is the price to buy (ask for you)  or sell (bid for you)  at a certain moment. Those prices by definition do not have a high or low. The orderbook gives the next near the last offered bid/ask.  Depending on the depth of the orderbook you can get more prices. Picture of the BTC/USD orderbook: https://www.cryptocompare.com/coins/btc/orderbook/USD

As soon as you are trying to determine a OHLC for bid/ask quotes you introduce a timeframe because your are aggregating data. The historical OHLC records offered by OANDA start with a timeframe of 5-seconds (S5). For all OHLC-records granularities you can specify the type: M/B/A (Mid/Bid/Ask), https://developer.oanda.com/rest-live-v20/instrument-ep/#collapse_2_example_curl_2

What is it you miss  in this data?

**Reply by jjphung** [reply]
Oh perfect! Your reply in that thread was exactly what I was missing and looking for. Fabricating OHCL from streams. I'll be sure to check it out. Thanks again!
